addons/fleet/views/fleet_vehicle_views.xml

```diff
@@ -106,7 +106,8 @@
                                 <field class="o_fleet_odometer_unit" name="odometer_unit"/>
                                 <button name="action_open_odometer_report" type="object" class="fa fa-line-chart me-1" title="Odometer Report"/>
                             </div>
-                            <field name="manager_id" domain="[('share', '=', False), ('company_id', '=', company_id)]" widget="many2one_avatar_user"/>
+                            <field name="manager_id" domain="[('share', '=', False), ('company_id', '=', company_id), ('all_group_ids', 'in', %(fleet.fleet_group_manager)d)]"
```

Comment:
This is probably due to the fact that the field domain is erased in the view.

```py
    manager_id = fields.Many2one(
        'res.users', 'Fleet Manager',
        domain=lambda self: [('all_group_ids', 'in', self.env.ref('fleet.fleet_group_manager').id), ('company_id', 'in', self.env.companies.ids)],
    )
```

In my opinion, it would be better to add ``('share', '=', False)`` in the field definition, and remove the domain in the view, so that this would be global instead of just being specific to this view only.

Could you please try my suggestion ? 

Thanks.